lib/ansible/modules/system/user.py

```diff
@@ -1813,6 +1866,24 @@ def modify_user_usermod(self):
 
         return (rc, out, err)
 
+    def user_info(self):
+        info = super(SunOS, self).user_info()
+        if info:
+            info += self._user_attr_info()
+        return info
+
+    def _user_attr_info(self):
+        info = [''] * 3
+        with open(self.USER_ATTR, 'rb') as file_handler:
+            for line in file_handler:
+                line = to_native(line.strip())
```

Comment:
If the file isn't open in binary mode, there is no reason to decode here.
```suggestion
                line = line.strip()
```